### ISSUE TITLE
Enabling non-free modules in iOS builds

### ISSUE THREAD

**Issue by akashivskyy** [opencv/opencv#13417]
I need to build OpenCV iOS framework with contrib and with non-free modules **enabled**. Currently, there's no possibility of passing additional CMake flags (like `-DOPENCV_ENABLE_NONFREE=ON`) to `build_framrwork.py`.

There are two proposed solutions to this:

1. Add a parameter `--enable_nonfree` to `build_framework.py` which would add the necessary flag to CMake invocation (would be off by default):

    ```sh
    ./platforms/ios/build_framework.py --enable_nonfree
   ```

   The advantage of this solution is that it's discoverable and pretty self-explanatory.

    The disadvantage is that it covers only one flag — if someone would want to map another CMake flag, they would need to modify build_framework once again.

3. Add a parameter `--cmake_flags` to `build_framework.py` which would allow to specify any flags that will be appended to CMake invocation (would be empty by default):

    ```sh
    ./platforms/ios/build_framework.py --cmake_flags="-DOPENCV_ENABLE_NONFREE=ON"
   ```

    The advantage of this solution is its versatility, users may pass any flag they want there.

    The disadvantage is that it might be misused — flags might be misspelled, conflict with the one used by the script and general number of build failures might increase because of that.

What's your opinion on this?

I will happily implement either solution according to your preference, or none of them if you have any objections or know a better way to handle the problem.

#### System information

- OpenCV: 4.0.0
- Platform: macOS 10.14.1
- Compiler: Xcode 10.1 10B61

**Comment by alalek** [comment]
I have no objections.
Feel free to add both into 3.4 branch. We will merge changes from 3.4 into master weekly/bi-weekly.

**Comment by akashivskyy** [comment]
This somehow wasn't closed after merging #13424 although it should be. 🤷‍♂️
